### ISSUE TITLE
Align generation of coins to market demands

### ISSUE THREAD

**Issue by grecinto** [dogecoin/dogecoin#2225]
Background: Over generation of coins causes dilution of the Dogecoin's value. We should align the generation with the demand. I.e. - when demand is high, allow generation of coins so we don't run out, thus, support expansion of economies and usage of Dogecoin in transactions/e-commerce. And taper down generation of coins(to zero if required) when usage scales down.

Ideal Solution: Track total "sold" and "unsold" coins and adjust the difficulty level in mining so the level goes high if the amount of unsold coins reach a certain maximum threshold. And reduce the level if the unsold coins count reach a certain minimum level.

Alternative Solution: Bring back the "coin generation" limiter feature and make the maximum total amount of coins that can be generated to be readable from configuration. Thus, it can be adjusted manually to certain ceiling.

There is a huge business impact on being able to control the total amount of coins generated by the system. And it will impact valuation of the Dogecoin. Not sure if this had been requested before, but typed away anyway, in case it can help.

Regards,
-Gerardo

**Comment by Im-Mr-Chris** [comment]
I think of twin prime conjectures instantly.
I have yet to work out the problem (searching for something that can handle high computation like matlab).

I'm no data scientist or have any real understanding of theoretical mathematics.
From the understanding that I have on this concept....we can actually use this to our advantage and just code the bull and bear markets.  That should solve the issue with all the fud right? YT is beyond distracted.  Too many bots...its centralized so i'm pretty sure that's not allowed.
but then again...im just some random nerd on the internet lol

**Comment by grecinto** [comment]
YES, this will solve all the FUD right now. :)

And per solution details, it can be as follows:

- Introduce a globally accessible persistence API so every time "coins" are sold(in the Dogecoin sub-system that takes care of "coin selling or allocation"), this API will be invoked to sum up to new total "all coins sold".

- Then, add another API that will read this persisted "total coins sold" info and the "total coins generated" and return the difference of the two numbers. This number which represents total number of unsold coins can then be used in the mining difficulty algorithm so we can factor in, in the formula this number. 

- And lastly, as mentioned, the formula for the level of difficulty needs to be tweaked so the "remaining unsold coins" will be used to drive the difficulty level. This can be simple hard code, or can be using some statistical model (perhaps where the Matlab you mention kicks in).

Thanks for your reply. Yes, it sounds interesting if we can wip this up. And most importantly, it "may" cause Dogecoin to fly to the moon, per valuation.

**Comment by patricklodder** [comment]
> Over generation of coins causes dilution of the Dogecoin's value.

This is intentional. See for example #23.

I'm closing this. The centralized solution you are describing could definitely be an interesting experiment, but not as a Dogecoin Core implementation.

**Comment by grecinto** [comment]
Another detail I did NOT mention is, we can add feature on the AWS "nodes" hosting the APIs the security feature(let us call this as "sentinel" nodes), so, if in case number of miner "hosts" start to taper down as there are less coins generated(wait period 'til usage expands again), the AWS "sentinel" nodes can serve as the "security" checkers.

Thus, network security of Doge coin will not be lessened, as the number of "sentinel" nodes in AWS will be there in exchange.

I know this had been closed by Doge developer, but adding this comment to complete the proposal seems fine. :)

**Comment by grecinto** [comment]
> > Over generation of coins causes dilution of the Dogecoin's value.
> 
> This is intentional. See for example #23.
> 
> I'm closing this. The centralized solution you are describing could definitely be an interesting experiment, but not as a Dogecoin Core implementation.

I think you had missed the business value of this proposition. Closed it too early, without waiting for the greater community and the leaders' input.

A feature such as this recommendation has never been coined, ever. It was not discussed as alternative in the cited #23 discussion. Just saying my opinion.

**Comment by delbonis** [comment]
With every new block the amount that the next block inflates the supply decreases.  In a few years the inflation rate will be eventually be below population growth rate.

The reason the issue was closed is because it's not possible to come to consensus on price or node count in a game-theoretically-secure way, so the only way for this mechanism to work would be relying on centralized services to act as oracles.  This does not align with the values or goals of the project, so there's not much point in continuing to discuss it here.

**Comment by grecinto** [comment]
The algorithm you discussed is not well advertised, thus, perhaps the reason why unlimited printing of coins is open for attacks in the investor side. If your team can publicize this then that should address the issue of being “not well understood”.

Per it cannot be determined required node count and making the Nodes “Oracle”, I supplied the additional idea after the issue was closed.

And thus, your answer is traveling into the future after the issue was closed, and justifying the quick action to close it. But the idea was provided after the fact.

And plus, computer systems are designed, this proposed feature expansion was seen not to have any principle violation. For this matter.
